src/main/java/org/opensearch/ad/task/ADBatchTaskRunner.java

```diff
@@ -261,32 +253,31 @@ private ActionListener<String> getTopEntitiesListener(
             adTaskCacheManager.setTopEntityInited(detectorId);
             int totalEntities = adTaskCacheManager.getPendingEntityCount(detectorId);
             logger.info("Total top entities: {} for detector {}, task {}", totalEntities, detectorId, taskId);
-            hashRing.getNodesWithSameLocalAdVersion(dataNodes -> {
-                int numberOfEligibleDataNodes = dataNodes.length;
-                // maxAdBatchTaskPerNode means how many task can run on per data node, which is hard limitation per node.
-                // maxRunningEntitiesPerDetector means how many entities can run per detector on whole cluster, which is
-                // soft limit to control how many entities to run in parallel per HC detector.
-                int maxRunningEntitiesLimit = Math
-                    .min(totalEntities, Math.min(numberOfEligibleDataNodes * maxAdBatchTaskPerNode, maxRunningEntitiesPerDetector));
-                adTaskCacheManager.setDetectorTaskLaneLimit(detectorId, maxRunningEntitiesLimit);
-                // scale down HC detector task slots in case there is less top entities in detection date range
-                int maxRunningEntities = Math.min(maxRunningEntitiesLimit, adTaskCacheManager.getDetectorTaskSlots(detectorId));
-                logger
-                    .debug(
-                        "Calculate task lane for detector {}: totalEntities: {}, numberOfEligibleDataNodes: {}, maxAdBatchTaskPerNode: {}, "
-                            + "maxRunningEntitiesPerDetector: {}, maxRunningEntities: {}, detectorTaskSlots: {}",
-                        detectorId,
-                        totalEntities,
-                        numberOfEligibleDataNodes,
-                        maxAdBatchTaskPerNode,
-                        maxRunningEntitiesPerDetector,
-                        maxRunningEntities,
-                        adTaskCacheManager.getDetectorTaskSlots(detectorId)
-                    );
-                forwardOrExecuteADTask(adTask, transportService, listener);
-                // As we have started one entity task, need to minus 1 for max allowed running entities.
-                adTaskCacheManager.setAllowedRunningEntities(detectorId, maxRunningEntities - 1);
-            }, listener);
+            DiscoveryNode[] dataNodes = { clusterService.localNode() };
```

Comment:
Can we add the TODO link for hashring issue here and also a comment stating that dataNodes will change when we will have multi node support?

Reply:
Sure, Ill add back the original code in places that I removed the hashring and Ill comment these out and tag the link

Reply:
+1 to @owaiskazi19